core/blockchain/impl/block_tree_impl.cpp

```diff
@@ -149,12 +176,64 @@ namespace kagome::blockchain {
     OUTCOME_TRY(storage_->putJustification(justification, block, node->depth));
 
     // update our local meta
-    tree_meta_->last_finalized = *node;
     node->finalized = true;
-    // TODO(kamilsa) PRE-355: Fix prune. Now prune removes all blocks that are
-    // higher than last finalized block. Instead we should remove only blocks
-    // that are not descendants of the last finalized block
-    // OUTCOME_TRY(prune());
+
+    // prune
```

Comment:
Why not extract it to a separate method? It's a piece of code that performs a single particular operation and is quite separate.

Reply:
Even two methods, collectChildrenForRemove and prune.

Reply:
Yep, it possible, but extract as internal (private) method. This mechanic using only as part of finalization, and neverwhere else.

Reply:
Even though it is used only once, extracting it to a separate method will make **finalize** method much more readable. So I suggest to extract it.

Reply:
Done